src/backend/distributed/connection/shared_connection_stats.c

```diff
@@ -204,7 +223,7 @@ GetMaxSharedPoolSize(void)
 {
 	if (MaxSharedPoolSize == ADJUST_POOLSIZE_AUTOMATICALLY)
 	{
-		return MaxConnections;
+		return GetMaxClientConnections();
```

Comment:
One interesting side-effect of this is that:
1) Once we already have `max_client_connections` reached via regular users
2) A superuser can connect to the node as a client, but cannot run any distributed queries (as we are blocked on the condition variable as we hit GetMaxClientConnections() outgoing connections as well -- most likely)

I wonder if we should have `superuser() ? MaxConnectios ? GetMaxClientConnections()` here such that superuser can have more control

Reply:
Fixed in GetMaxClientConnections itself